drivers/comparator/qm_comparator.c

```diff
@@ -79,6 +81,9 @@ int qm_ac_set_config(const qm_ac_config_t *const config)
 	QM_SCSS_CMP->cmp_ref_pol = config->polarity;
 	QM_SCSS_CMP->cmp_pwr = config->power;
 
+	/* Provide some delay after power Change to Comparator*/
+	clk_sys_udelay(5 * ONE_MICROSECOND);
```

Comment:
@youvedeep can you please try removing the delay and using a read back on the QM_SCSS_CMP->cmp_pwr register instead?

So, you will end up with a register write followed by a register dummy read, but no delays.
For a dummy read all you need is a line like

`QM_SCSS_CMP->cmp_pwr;`